lib/code_corps/policy/helpers.ex

```diff
@@ -86,9 +86,10 @@ defmodule CodeCorps.Policy.Helpers do
   @doc """
   Retrieves task from associated record
   """
-  @spec get_task(Changeset.t | TaskSkill.t | UserTask.t) :: Task.t
+  @spec get_task(Changeset.t | TaskSkill.t | UserTask.t | map) :: Task.t
   def get_task(%TaskSkill{task_id: task_id}), do: Repo.get(Task, task_id)
   def get_task(%UserTask{task_id: task_id}), do: Repo.get(Task, task_id)
+  def get_task(%{"task_id" => task_id}), do: Repo.get(Task, task_id)
```

Comment:
Why do we need a map here when we didn't before?

Reply:
Yeah good call out.  So this was a result of the `create` action sending through a map instead of the struct....after removing the use of ja_resource, the policy accepts a `map` for the `create` action

Reply:
Ah, okay. That's a little unfortunate in the sense that we lose the pattern matching against `%Task{}`, for example, unless I'm misunderstanding the change.

Reply:
I'm not sure there's much we can or _should_ do to change that, but just something I noticed.

Reply:
>That's a little unfortunate in the sense that we lose the pattern matching against `%Task{}`, for example, unless I'm misunderstanding the change.

We don't, not really.  We never pattern matched against `%Task{}` when authorizing an action coming from a `UserTaskController`. We pattern-matched on a `%UserTask{}` or a `%Changeset{}`, containing a `:task_id` change. Instead, we now pattern-match on a `%UserTask{}` or a `param` map  containing a `"task_id"` key.

We were, and still are, pattern-matching on a  `%Task{}` on a change coming from the `TaskController.`

I'm actually thinking I initially went a bit too far with trying to dry up the code here. We have this `get_task` method which is called from everywhere, implying there's a shared approach in authorizing. 

Really,  when authorizing actions on a `UserTask`, it's pretty simple.

- for `:update` and `:delete`, get the `user_task` record and associated `task` record, decide if the `current_user` can do those things
- for `:create`, look at the `params` sent from the client and decide if the `current_user` can create such a record 

Maybe we should make it a bit less dry to improve clarity. Not something that's the concern of this PR, though.

I do like the move away from changesets, though. They're intended for validation and we were forcing them into authorization.

Previously, it was

- get resource/params
- init changeset with resource/params
- authorize action on resource/changeset
- init same changeset again to validate an perform action

Not it's 

- get resource/params
- authorize action on resource/params
- perform action

@snewcomer Again, this is a conversation about what we could potentially do. Your changes are in line with the approach we're using to move away in other places.